google/cloud/storage/client.h

```diff
@@ -640,8 +715,11 @@ class Client {
    *     `IfMetagenerationNotMatch`, `Projection`, and `UserProject`.
    *
    *
-   * @throw std::runtime_error if the metadata cannot be fetched using the
-   * current policies.
+   * @throw std::runtime_error if there is a permanent failure, or if there were
+   *     more transient failures than allowed by the current retry policy.
+   *
+   * @par Idempotency
+   * Being a read-only operation this is always idempotent.
```

Comment:
Is this true when the metadata changes? I would expect a metageneration precondition.

Reply:
Under the wikipedia definition this is true.  Read-only operations do not change the state (lucky for us there is no "last read" timestamp), so all read-only operations are idempotent.